integration_tests/steps/device.py

```diff
@@ -0,0 +1,57 @@
+from behave import given, when, then
+from os import system
+from time import sleep
+from integration_tests.test_load_connection import AutoBidOnLoadDevice
+from integration_tests.test_pv_connection import AutoOfferOnPVDevice
+
+
+@given('redis container is started')
+def step_impl(context):
+    system('docker run -d -p 6379:6379 --name redis.container -h redis.container '
+           '--net integtestnet gsyd3a/d3a:redis-staging')
+
+
+@given('d3a container is started using setup file {setup_file}')
+def step_impl(context, setup_file):
+    sleep(3)
+    system(f'docker run -d --env REDIS_URL=redis://redis.container:6379/ --net integtestnet -t d3a '
+           f'run -t 1s -s 15m --setup {setup_file}')
+
+
+@when('the external client container is started with test_load_connection')
+def step_impl(context):
+    # Wait for d3a to activate all areas
+    sleep(5)
+    # Connects one client to the load device
+    context.load = AutoBidOnLoadDevice('load', autoregister=True, redis_url='redis://localhost:6379/')
+
+
+@when('the external client container is started with test_pv_connection')
+def step_impl(context):
+    # Wait for d3a to activate all areas
+    sleep(5)
+    # Connects one client to the load device
+    context.load = AutoOfferOnPVDevice('pv', autoregister=True, redis_url='redis://localhost:6379/')
```

Comment:
May be offline, if you can show me where exactly is `the external client container is started` ?

Reply:
The word container is indeed misleading here. I am removing this to change the step name to `the external client is started with test_pv_connection`.